prinz/src/main/scala/pl/touk/nussknacker/prinz/util/collection/immutable/VectorMultimapUtils.scala

```diff
@@ -1,12 +1,13 @@
-package pl.touk.nussknacker.prinz.pmml.model
+package pl.touk.nussknacker.prinz.util.collection.immutable
 
-import pl.touk.nussknacker.prinz.util.collection.immutable.VectorMultimap
+import scala.collection.mutable
 
 object VectorMultimapUtils {
+
   implicit class VectorMultimapAsRowset[K, V](val inputsMap: VectorMultimap[K, V]) {
-    def forEachRow[T](f: Map[K, V] => T) : IndexedSeq[T] = {
+    def forEachRow[T](f: Map[K, V] => T): IndexedSeq[T] = {
       val totalTuples = inputsMap.map(_._2.length).max
-      val iteratorsMap = inputsMap.mapVectors(_.iterator)
+      val iteratorsMap = mutable.Map() ++ inputsMap.mapVectors(_.iterator)
```

Comment:
why do you need mutable.Map() here?

Reply:
I missed that earlier. With immutable map, iterators do not change and each time `f` receives the first row as argument.